src/modules/sync/components/sync-dialog.tsx

```diff
@@ -18,60 +19,101 @@ import { useDefaultSyncTree } from 'src/modules/sync/hooks/use-default-sync-tree
 import { useSyncDialogTexts } from 'src/modules/sync/hooks/use-sync-dialog-texts';
 import { getSiteEnvironment } from 'src/modules/sync/lib/environment-utils';
 import { useI18nLocale } from 'src/stores';
+import { useLatestRewindId, useRemoteFileTree } from 'src/stores/sync';
+import { TreeViewLoadingSkeleton } from './tree-view-loading-skeleton';
 import type { SyncSite } from 'src/hooks/use-fetch-wpcom-sites/types';
 
 type SyncDialogProps = {
 	type: 'push' | 'pull';
 	localSite: SiteDetails;
 	remoteSite: SyncSite;
-	onSubmit: ( syncData: TreeNode[] ) => void;
+	onPush: ( syncData: TreeNode[] ) => void;
+	onPull: ( syncData: TreeNode[], rewindId: string ) => void;
```

Comment:
We don't use it here https://github.com/Automattic/studio/pull/1596/files#diff-70c748995b4acd847cec8c7e9128ad620a6f97c16cc4305109a5703e667761dcR164
```suggestion
	onPull: ( syncData: TreeNode[] ) => void;
```

Reply:
Done https://github.com/Automattic/studio/pull/1596/commits/cc43a36ae31734771c524403c567d79e7a52329f